docs/content/guides/dagster-pipes/modifying-existing-code-to-work-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,270 @@
+src="/images/guides/dagster-pipes/modifying-code-1-materialize-dagster-assets.png"
+width={1328}
+height={836}
+/>
+
+### Monitor code execution in Dagster UI
+
+To monitor the run, navigate to the [Run details page](/concepts/webserver/ui#run-details). This page includes timing, raw compute logs, and structured events. The bottom section shows filterable logs from the run, detailing events, their types, and specifics. There are two types of log: structured event and raw compute.
+
+#### Structured event logs
+
+Structured logs are enriched and categorized with metadata. For example, a label of what the log entry is about, links to an asset’s metadata, and the event type. This detailed structuring not only facilitates easier filtering and searching within the logs, but also populates the asset catalog and status information in the asset graph. We’ll come back to how you can log metadata from your existing code to Dagster in a little bit.
+
+<Image
+alt="Structured event logs"
+src="/images/guides/dagster-pipes/modifying-code-2-structured-event-logs.png"
+width={1524}
+height={844}
+/>
+
+#### Raw compute logs
+
+If your existing code produces stdout or stderr logs, like a `print` statement in our sample Python script, Dagster will display these in the UI's raw compute log view. To see the `stdout` log, toggle the log section to **stdout**.
+
+<Image
+alt="Raw compute logs"
+src="/images/guides/dagster-pipes/modifying-code-3-raw-compute-logs.png"
+width={1498}
+height={854}
+/>
+
+---
+
+## Stream information back to Dagster
+
+### Make Dagster context available in existing code
+
+Getting existing code to work with Dagster Pipes requires adding a few lines of code:
+
+- Imports from `dagster-pipes`
+
+- A call that connects to Dagster Pipes: <PyObject module="dagster_pipes" object="open_dagster_pipes"/> initializes the Dagster Pipes context that can be used to stream information back to Dagster. It is recommended to call this function near the entry point of a pipes process.
+
+  <Note>
+
+  The with <code>with open_dagster_pipes() as context:</code> is a context manager in Python, ensuring resource setup and cleanup for a specific segment of code. It's useful for tasks requiring initial setup and final teardown, like opening and closing connections. In this case, the context manager is used to initialize and close the Dagster Pipes connection.
+
+  </Note>
+
+- An instance of the Dagster Pipes context via <PyObject module="dagster_pipes" object="PipesContext" method="get"/>.
+
+In our sample Python script, the changes would look like the following:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_with_context.py
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    ...
```

Comment:
the ... here is a bit odd, especially since it's replacing a single `print` line. I wonder if turning this section into steps (1. Make Dagster context available in existing code, 2. Send log messages to Dagster, 3...) would make it a bit better? The ordering isn't quite exact though because you just need to do step 1 and then you can choose which of the following steps you want....